system/Database/BaseConnection.php

```diff
@@ -1526,7 +1526,7 @@ public function isWriteType($sql): bool
      *
      * Must return an array with keys 'code' and 'message':
      *
-     *  return ['code' => null, 'message' => null);
+     * @return array{code: int|string|null, message: string|null}
```

Comment:
Is this legit doc code with VS Studio, etc? @paulbalandan

Reply:
Good catch. VS Code does not understand array shapes yet.
```suggestion
     * @return array<string, int|string|null>
     * @phpstan-return array{code: int|string|null, message: string|null}
```